README.md

```diff
@@ -159,47 +83,77 @@ useFlagsStatus retrieves the ready state and error events.
 Follow the following steps in order to delay rendering until the flags have been fetched.
 
 ```jsx
-import { useFlagsStatus } from '@unleash/proxy-client-react'
+import { useFlagsStatus } from '@unleash/proxy-client-react';
 
 const MyApp = () => {
   const { flagsReady, flagsError } = useFlagsStatus();
 
   if (!flagsReady) {
-    return <Loading />
+    return <Loading />;
   }
-  return <MyComponent error={flagsError}/>
-}
-
+  return <MyComponent error={flagsError} />;
+};
 ```
 
 ## Updating context
 
 Follow the following steps in order to update the unleash context:
 
 ```jsx
-import { useUnleashContext, useFlag } from '@unleash/proxy-client-react'
+import { useUnleashContext, useFlag } from '@unleash/proxy-client-react';
 
 const MyComponent = ({ userId }) => {
-  const variant = useFlag("my-toggle");
+  const variant = useFlag('my-toggle');
   const updateContext = useUnleashContext();
 
   useEffect(() => {
     // context is updated with userId
-    updateContext({ userId })
-  }, [userId])
+    updateContext({ userId });
+  }, [userId]);
 
   useEffect(() => {
     async function run() {
-    // Can wait for the new flags to pull in from the different context
+      // Can wait for the new flags to pull in from the different context
       await updateContext({ userId });
       console.log('new flags loaded for', userId);
     }
     run();
   }, [userId]);
-}
-
+};
 ```
 
+# Advanced use cases
+
+## Deferring client start
+
+By default, the Unleash client will start polling the Proxy for toggles immediately when the `FlagProvider` component renders. You can prevent it by setting `startClient` prop to `false`. This is useful when you'd like to for example bootstrap the client and work offline.
+
+Deferring the client start gives you more fine-grained control over when to start fetching the feature toggle configuration. This could be handy in cases where you need to get some other context data from the server before fetching toggles, for instance.
+
+To start the client, use the client's `start` method. The below snippet of pseudocode will defer polling until the end of the `asyncProcess` function.
+
+```jsx
+const client = useRef(
+  new UnleashClient({
+    /* ... */
+  })
+);
+
+useEffect(() => {
+  const asyncProcess = async () => {
+    // do async work ...
+    client.current.start();
+  };
+  asyncProcess();
+}, []);
+
+return (
+  // Pass client as `unleashClient` and set `startClient` to `false`
+  <FlagProvider unleashClient={client.current} startClient={false}>
+    <App />
+  </FlagProvider>
+);
+```
```

Comment:
Seems to be related to a change we haven't decided on/merged/released yet. It's not the current way of initializing.

Reply:
Not exactly. In React it's better to store external state in refs in components instead of global scope.

Reply:
> Not exactly. In React it's better to store external state in refs in components instead of global scope.

Can you point me to some documentation for this pattern as a best practice? I haven't seen this pattern before.